src/model/index.ts

```diff
@@ -289,19 +289,37 @@ export class Model {
             || !! this.bookmarks.node(id as Bookmarks.NodeID));
     }
 
-    /** Put the set of currently-selected items in the current window. */
-    async putSelectedInWindow(options: {copy: boolean}) {
-        await this.putItemsInWindow({
-            items: copyIf(options.copy, Array.from(this.selectedItems())),
+    /** Put the set of currently-selected items in the specified folder
+     * when the toFolderId option is set, otherwise the current window.
+     *
+     * Note: When copying is disabled, the source items will be deselected. */
+    async putSelectedIn(
+        options?: {
+            copy?: boolean, toFolderId?: Bookmarks.NodeID,
+        }) {
+        const from_items = Array.from(this.selectedItems());
+        const items = copyIf(options?.copy === true, from_items);
+
+        if (!options?.copy) {
+            from_items.forEach(i => i.$selected = false);
+        }
```

Comment:
I think I'd wait to do this until after `putItemsIn...()` for two reasons. First, if `putItemsIn...()` throws for some reason, the original selection will be left intact. And second, it looks a little less confusing--sometimes there's a noticeable delay when moving a large number of items. Leaving the original items selected will show the items moving along with their selection status, so it's a little easier to tell where the items are being placed.

Reply:
Makes sense!